portfolio/src/main/java/com/google/sps/servlets/AddCommentServlet.java

```diff
@@ -38,34 +38,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String email = userService.getCurrentUser().getEmail();
     String message = request.getParameter("message");
     long timestamp = System.currentTimeMillis();
-    String image = getUploadedFileUrl(request, "image");
+    Optional<String> image = getUploadedFileUrl(request);
 
     Entity newComment = new Entity("Comment");
 
     newComment.setProperty("email", email);
     newComment.setProperty("timestamp", timestamp);
     newComment.setProperty("message", message);
-    newComment.setProperty("image", image);
+    image.ifPresent(url -> newComment.setProperty("image", url));
 
     DatastoreServiceFactory.getDatastoreService().put(newComment);
   }
 
   /** Returns the URL to the uploaded file or null if the user didn't upload a file. */
-  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+  private Optional<String> getUploadedFileUrl(HttpServletRequest request) {
     BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
-    List<BlobKey> allCommentImageKeys = blobstoreService.getUploads(request).get("image");
+    Optional<List<BlobKey>> allCommentImageKeys = Optional.ofNullable(blobstoreService.getUploads(request).get("image"));
 
     // User submitted form without selecting a file (on dev server)
-    if (allCommentImageKeys == null || allCommentImageKeys.isEmpty()) return null;
+    if (!allCommentImageKeys.isPresent() || allCommentImageKeys.get().isEmpty()) return Optional.empty();
```

Comment:
Yeah, I understand your chat message now. It's not totally necessary to convert allCommentImageKeys to an Optional here since you are just going to return if it's empty. Having to ```get()``` it every time is pretty cumbersome too. You're call if you want to keep it this way.